### ISSUE TITLE
PropertyGrid: objects not expandable

### ISSUE THREAD

**Issue by RussKie** [dotnet/winforms#1553]
* Have you experienced this same bug with .NET Framework?: No

**Problem description:**

The property grid does not allow to expand complex objects if those aren't at the top level of object hierarchy.

![image](https://user-images.githubusercontent.com/4403806/62362075-e5c73a80-b524-11e9-8db6-cb9fb62431a0.png)

**Actual behavior:**

![image](https://user-images.githubusercontent.com/4403806/62362233-3a6ab580-b525-11e9-97e4-50c4d4937e25.png)
![image](https://user-images.githubusercontent.com/4403806/62362325-5ec69200-b525-11e9-941e-d911add3681d.png)

**Expected behavior:**

Object at all levels are expandable.

**Minimal repro:**

Run WinFormsControlsClassicTests and WinformsControlsTest side-by-side, click [Collection Editors] button and compare the property grids.

**Comment by zsd4yr** [comment]
@DustinCampbell, @AdamYoblick and I feel like you mentioned something here -- do you have any thoughts?

**Comment by weltkante** [comment]
`ComponentConverter` should be the one responsible for expanding in the property grid (it overrides GetProperties, same as ExpandableObject does). I don't know how ComponentConverter is linked to the `Component` base class, its not an attribute on the type. Maybe that link is broken in .NET Core.

**Comment by DustinCampbell** [comment]
> @DustinCampbell, @AdamYoblick and I feel like you mentioned something here -- do you have any thoughts?

This is about the .NET Core property grid and not the framework version, which gets used in VS. So, I don't really have anything to add.

**Comment by RussKie** [comment]
Thank you @hughbe 🚀

**Comment by danmosemsft** [comment]
reopening since apparently this is desired for 3.0 GA which means it would need a port PR.

cc @safern.

**Comment by danmosemsft** [comment]
(if not, please re-close )

**Comment by zsd4yr** [comment]
@danmosemsft 	will someone cherry pick the change in corefx master into the 3.0 GA branch or should @hughbe 	make that PR?

**Comment by danmosemsft** [comment]
@RussKie perhaps this can be closed - maybe after our code flows and you've validated ?

**Comment by RussKie** [comment]
@danmosemsft this looks like fixed in 5.0 but 3.1 is still broken. Is there anything we can do to get it fixed in 3.1 as well?

* 3.1 Preview1
![image](https://user-images.githubusercontent.com/4403806/66021910-95258b00-e52f-11e9-8f0d-e3f7ae489430.png)


* 5.0
![image](https://user-images.githubusercontent.com/4403806/66021180-0fa0db80-e52d-11e9-90ed-4053d9cff522.png)

**Comment by weltkante** [comment]
Seems to work for me in 3.0 GA, can't test 3.1

<details><summary>example</summary>

```C#
public partial class Form1 : Form
{
    public Form1()
    {
        InitializeComponent();

        PropertyGrid pg;
        Controls.Add(pg = new PropertyGrid() { Dock = DockStyle.Fill });
        pg.SelectedObject = new Dummy();
    }

}

public class Dummy
{
    public ListView MyListView { get; } = new ListView();
}
```

![image](https://user-images.githubusercontent.com/5845814/66026817-e6b32300-e4f9-11e9-890b-f8a5032ae1cb.png)

</details>

**Comment by safern** [comment]
@RussKie the fix seems to be in the 3.1 branch... have you guys consumed the latest Microsoft.NETCore.App from 3.1? Would you mind checking what version you tested agains?

Here's the source code for 3.1: https://github.com/dotnet/corefx/blob/release/3.1/src/System.ComponentModel.TypeConverter/src/System/ComponentModel/ReflectTypeDescriptionProvider.cs#L139

**Comment by RussKie** [comment]
Thank you for the confirmation.
I have created brand new apps targeting 3.0 and 3.1 outside the repo, and it both cases worked.
